packages/hardware-ledger/src/LedgerKeyAgent.ts

```diff
@@ -148,98 +165,207 @@ export class LedgerKeyAgent extends KeyAgentBase {
+        transport = await TransportWebUSB.open(device);
+      } else {
+        throw new errors.TransportError(`Invalid device object provided for communication type ${communicationType}`);
+      }
+    } catch (error) {
+      throw new errors.TransportError('Failed to open a transport for a given device', error);
+    }
+    this.attachDisconnectionCleanupHandler(transport);
+    return transport;
+  }
+
   /**
    * @throws TransportError
    */
   static async createTransport({
     communicationType,
-    activeTransport,
-    devicePath = ''
+    nodeHidDevicePath = ''
   }: CreateLedgerTransportProps): Promise<LedgerTransportType> {
+    let transport: LedgerTransportType;
     try {
-      if (communicationType === CommunicationType.Node) {
-        return await TransportNodeHid.open(devicePath);
-      }
-      return await (activeTransport && activeTransport instanceof TransportWebHID
-        ? TransportWebHID.open(activeTransport.device)
-        : TransportWebHID.request());
+      transport =
+        communicationType === CommunicationType.Node
+          ? await TransportNodeHid.open(nodeHidDevicePath)
+          : await TransportWebUSB.request();
     } catch (error) {
       throw new errors.TransportError('Creating transport failed', error);
     }
+
+    this.attachDisconnectionCleanupHandler(transport);
+    return transport;
   }
 
   /**
    * @throws TransportError
    */
-  static async createDeviceConnection(activeTransport: LedgerTransport): Promise<LedgerConnection> {
+  private static async testConnection(activeConnection: LedgerConnection): Promise<void> {
     try {
-      const deviceConnection = new LedgerConnection(activeTransport);
       // Perform app check to see if device can respond
-      await deviceConnection.getVersion();
-      return deviceConnection;
+      await activeConnection.getVersion();
     } catch (error) {
-      throw new errors.TransportError('Cannot communicate with Ledger Cardano App', error);
+      throw new errors.TransportError(CARDANO_APP_CONNECTION_ERROR_MESSAGE, error);
     }
   }
 
   /**
    * @throws TransportError
    */
+  static async createDeviceConnection(activeTransport: LedgerTransport): Promise<LedgerConnection> {
+    const deviceConnection = new LedgerConnection(activeTransport);
+    await this.testConnection(deviceConnection);
+    return deviceConnection;
+  }
+
+  private static parseNodeHidDevicePathOrDeviceParam(nodeHidDevicePathOrDevice?: string | LedgerDevice) {
+    let device: LedgerDevice;
+    let nodeHidDevicePath: string;
+
+    if (nodeHidDevicePathOrDevice instanceof HID || isUsbDevice(nodeHidDevicePathOrDevice)) {
+      device = nodeHidDevicePathOrDevice;
+    }
+    if (typeof nodeHidDevicePathOrDevice === 'string') {
+      nodeHidDevicePath = nodeHidDevicePathOrDevice;
+    }
+
+    return {
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore TS2454 used before being assigned
```

Comment:
I think it's better to obey to ts rules rather than disabling

```suggestion
    let device: LedgerDevice;
    let nodeHidDevicePath: string | undefined;

    if (nodeHidDevicePathOrDevice instanceof HID || isUsbDevice(nodeHidDevicePathOrDevice)) {
      device = nodeHidDevicePathOrDevice;
    }
    if (typeof nodeHidDevicePathOrDevice === 'string') {
      nodeHidDevicePath = nodeHidDevicePathOrDevice;
    }

    if (!device) throw ...
    
    return {
```

Reply:
I totally agree about obeying the rules. I just didn't come up with your solution 😉 
Regarding that throw you added - it is expected that both variables will stay undefined.

Reply:
Re: throw - Added a throw in case the user passed an invalid parameter against TS. In such case, it would fail anyway on further implementation but this approach is better.

Reply:
I improved it. To make it descriptive I embed the method name that the error message relates to. In order to ensure the message stays accurate in case the method gets renamed I created [a variable](https://github.com/input-output-hk/cardano-js-sdk/pull/1161/files#diff-19303657c074c5ebc3361bdd09aee5cf4d2c2e5b7de7a019d5bdb04f249e20bdR100) for that name and I am using it to [create the method](https://github.com/input-output-hk/cardano-js-sdk/pull/1161/files#diff-19303657c074c5ebc3361bdd09aee5cf4d2c2e5b7de7a019d5bdb04f249e20bdR310-R323) and in [the error message](https://github.com/input-output-hk/cardano-js-sdk/pull/1161/files#diff-19303657c074c5ebc3361bdd09aee5cf4d2c2e5b7de7a019d5bdb04f249e20bdR120). Let me know it this approach is fine for you.